src/EditorFeatures/CSharpTest/CodeActions/ConvertLinq/ConvertLinqQueryToForEachTests.cs

```diff
@@ -0,0 +1,2149 @@
+                    yield return n1;
+                }
+            }
+        }
+
+        var q = localFunction();
+        foreach (var b in q)
+        {
+            Console.WriteLine(b);
+        }
+    }
+}
+";
+
+            await TestInRegularAndScriptAsync(source, output);
+        }
+
+        [Fact]
+        public async Task UsageInForEachSameVariableName()
+        {
+            string source = @"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+class C
+{
+    void M(IEnumerable<int> nums)
+    {
+        var q = [||]from int n1 in nums 
+                from int n2 in nums
+                select n1;
+        foreach(var n1 in q)
+        {
+            Console.WriteLine(n1);
+        }
+    }
+}
+";
+
+            string output = @"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+class C
+{
+    void M(IEnumerable<int> nums)
+    {
+        IEnumerable<int> localFunction()
+        {
+            foreach (int n1 in nums)
+            {
+                foreach (int n2 in nums)
+                {
+                    yield return n1;
+                }
+            }
+        }
+
+        var q = localFunction();
+        foreach(var n1 in q)
+        {
+            Console.WriteLine(n1);
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(source, output);
+        }
+
+        [Fact]
+        public async Task LinqDefinedInForEach()
+        {
+            string source = @"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+class C
+{
+    void M(IEnumerable<int> nums)
+    {
+        foreach(var b in [||]from int n1 in nums 
+                from int n2 in nums
+                select n1)
+        {
+            Console.WriteLine(b);
+        }
+    }
+}
+";
+
+            string output = @"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+class C
+{
+    void M(IEnumerable<int> nums)
+    {
+        foreach (int n1 in nums)
+        {
+            foreach (int n2 in nums)
+            {
+                var b = n1;
+                Console.WriteLine(b);
+            }
+        }
+    }
+}
+";
+
+            await TestInRegularAndScriptAsync(source, output);
+        }
+
+        [Fact]
+        public async Task LinqDefinedInForEachSameVariableName()
+        {
+            string source = @"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+class C
+{
+    void M(IEnumerable<int> nums)
+    {
+        foreach(var n1 in [||]from int n1 in nums 
+                          from int n2 in nums
+                          select n1)
+        {
+            Console.WriteLine(n1);
+        }
+    }
+}
+";
+
+            string output = @"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+class C
+{
+    void M(IEnumerable<int> nums)
+    {
+        foreach (int n1 in nums)
+        {
+            foreach (int n2 in nums)
+            {
+            Console.WriteLine(n1);
+        }
```

Comment:
indentation.

Reply:
@heejaechang@outlook.com, could you please help?

---
In reply to: [175267336](https://github.com/dotnet/roslyn/pull/25362#discussion_r175267336) [](ancestors = 175267336)

Reply:
setting format annotation on block or { } tokens doesn't work? I  believe it is happening because none of "{", "}", "C...;" is generated or have formatting annotionats, so formatter is not touching them.

Reply:
Thank you! Just applied. It helped.

---
In reply to: [176885406](https://github.com/dotnet/roslyn/pull/25362#discussion_r176885406) [](ancestors = 176885406)